### ISSUE TITLE
How to get the public key ID/Fingerprint from a detatched signature

### ISSUE THREAD

**Issue by danhab99** [openpgpjs/openpgpjs#1325]
```ts
import * as openpgp from "openpgp";

let detatchedSignature = "-----BEGIN PGP SIGNATURE----- insert signature here -----END PGP SIGNATURE-----";
openpgp.readSignature({
  armoredSignature: detatchedSignature,
}).then((sig: openpgp.Signature) => {
  let signingPublicKey = // whaaa??
});

```

According the the [typedefs file](https://github.com/openpgpjs/openpgpjs/blob/master/openpgp.d.ts) the `verify` function needs the public key to test against, that makes sence. The problem is that I don't know what the public key might be. I have a database of public keys that I can query by `keyID` or `fingerprint`, and according to [RFC 4880](https://datatracker.ietf.org/doc/html/rfc4880) it is possible to get the fingerprint. The [typedefs for `openpgp.readSignature()`](https://github.com/openpgpjs/openpgpjs/blob/master/openpgp.d.ts#L109) I can input a detached signature and get the promise of a [`signature`](https://github.com/openpgpjs/openpgpjs/blob/master/openpgp.d.ts#L112), but the object doesn't provide the fingerprint.

So how can I get the fingerprint or ID of the public key that created a signature using OpenPGP.js? Do I need to ask the user for the public key too or can I just get it from the signature?

**Comment by twiss** [comment]
Hey :wave: It's indeed technically possible to get the key IDs or fingerprints from the signature, even though there's no public method for it at the moment. If you want to go about creating one, you could take https://github.com/openpgpjs/openpgpjs/blob/f0280262172ddc54910c056e2c8d8d42be5643c2/src/message.js#L97-L100 as inspiration, and make a similar method in the Signature class. (A PR would be welcome ^.^)

**Comment by larabr** [comment]
Fixed in #1331.
